tcmur_cmd_handler.c

```diff
@@ -33,14 +33,30 @@ static void _cleanup_spin_lock(void *arg)
 	pthread_spin_unlock(arg);
 }
 
-void tcmur_command_complete(struct tcmu_device *dev, struct tcmulib_cmd *cmd,
-			    int rc)
+void tcmur_tcmulib_cmd_complete(struct tcmu_device *dev,
+				struct tcmulib_cmd *cmd, int rc)
 {
 	struct tcmur_device *rdev = tcmu_dev_get_private(dev);
+	struct tcmur_cmd *tcmur_cmd = cmd->hm_private;
+	struct timespec curr_time;
 
 	pthread_cleanup_push(_cleanup_spin_lock, (void *)&rdev->lock);
 	pthread_spin_lock(&rdev->lock);
 
+	if (tcmur_cmd->timed_out) {
+		if (tcmur_get_time(dev, &curr_time)) {
+			tcmu_dev_info(dev, "Timed out command id %hu completed with status %d.\n",
+				      cmd->cmd_id, rc);
```

Comment:
If the tcmur_get_time() return non-zero, the dev's timeout feature will be disabled, is this expected ?

Reply:
It was.

Do you think it shouldn't be disabled? I do not care either way. It is coded like that just because it was simple. I was just thinking if it worked initially but failed here we are probably in big trouble and it does not matter.

Reply:
Yeah, it makes sense and I am okay with this :-)